### ISSUE TITLE
Segmentation fault on `next` in `split_workloads.more-or-less-multi-serial-mix` test

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#5070]
On commit 80a5e5162780dabc57cebb45dbb5e9303d6ed882 , compiling in release with rec++ on newton, I got the following seg fault in the `split_workloads.more-or-less-multi-serial-mix` test:

```C++
Version: rethinkdb 2.1.0-0BETA2-432-g80a5e5-dirty (CLANG 3.2 (tags/RELEASE_32/final))
error: Error in src/arch/runtime/thread_pool.cc at line 359:
error: Segmentation fault from reading the address 0x32.
error: Backtrace:
Version: rethinkdb 2.1.0-0BETA2-432-g80a5e5-dirty (CLANG 3.2 (tags/RELEASE_32/final))
error: Error in src/arch/runtime/thread_pool.cc at line 359:
Version: rethinkdb 2.1.0-0BETA2-432-g80a5e5-dirty (CLANG 3.2 (tags/RELEASE_32/final))
error: Segmentation fault from reading the address (nil).
error: Error in src/arch/runtime/thread_pool.cc at line 359:
error: Backtrace:
error: Segmentation fault from reading the address 0x21f.
error: Backtrace:
error: Mon Nov  9 20:01:59 2015
       
       1 [0xc843b0]: backtrace_t::backtrace_t() at backtrace.cc:203
       2 [0xc84ac3]: lazy_backtrace_formatter_t::lazy_backtrace_formatter_t() at basic_string.h:269
       3 [0xc83e14]: format_backtrace(bool) at backtrace.cc:197
       4 [0x7d9fd3]: report_fatal_error(char const*, int, char const*, ...) at basic_string.h:287
       5 [0x72d330]: linux_thread_pool_t::fatal_signal_handler(int, siginfo*, void*) at thread_pool.cc:359
       6 [0x7f796e410cb0]: /lib/x86_64-linux-gnu/libpthread.so.0(+0xfcb0) [0x7f796e410cb0] at 0x7f796e410cb0 (/lib/x86_64-linux-gnu/libpthread.so.0)
       7 [0x7655a0]: alt::page_cache_t::consider_evicting_all_current_pages(alt::page_cache_t*, auto_drainer_t::lock_t) at page_cache.cc:175
       8 [0x76de0b]: callable_action_instance_t<std::_Bind<void (*(alt::page_cache_t*, auto_drainer_t::lock_t))(alt::page_cache_t*, auto_drainer_t::lock_t)> >::run_action() at functional:1145
       9 [0x72bf88]: coro_t::run() at coroutines.cc:214
error: Exiting
```

**Comment by danielmewes** [comment]
Oh I think this is my fault. Working on a fix.

**Comment by danielmewes** [comment]
This is hard to reproduce (it didn't happen the next 20 times I ran the test). I think it only happens if a page gets evicted right when the "read ahead" phase of a table ends.
Anyway, the likely fix is in review 3318 by @Tryneus

**Comment by danielmewes** [comment]
Fixed in 1f968de
